wordpress-mobile/WordPress-Android
# Issue 12407/story title container focus

Fixes #12407 #12411

## Solution
The main focus of this PR is to fix all the issues that are related to the keyboard and the story title's `EditText` focus. 

1. You can now click on anywhere within the story title's layout to trigger the keyboard and focus c03e2d5
2. The keyboard now opens up when the bottom sheet is opened. 

## Testing

1. Create a story post. 
2. Click Next. 
3. See the bottom sheet open with the keyboard active and the story title focused.

<kbd><img src="https://user-images.githubusercontent.com/1509205/88044799-d36a5600-cb13-11ea-903f-48130ba2629e.gif" width="320"></kbd>

-------

1. Create a story post. 
2. Click Next. 
3. See the bottom sheet open with the keyboard active and the story title focused.
4. Close the keyboard. 
5. Click anywhere within the story title layout like below.

<kbd><img src="https://user-images.githubusercontent.com/1509205/88044650-c5b4d080-cb13-11ea-9262-8d9bd1359cdc.png" width="320"></kbd>

------- 
## Concerns

The issue I am having with this solution, which is probably unrelated to these changes but more related to this `StoryComposerActivity` is the way how we handle the window. We are currently using a style that makes the window full screen but that interferes with the keyboard's ability to adjust the view. i.e `adjustResize` and the `fullScreen` styles don't play well together. 

This is main part of the style that is doing this

```
        <item name="android:immersive">true</item>
        <item name="android:windowFullscreen">true</item>
```

So the issue that's taking place now is since we can't utilize the `android:windowSoftInputMode` everything is being done manually. We even have logic to [adjust the view when the keyboard is opened](https://github.com/wordpress-mobile/WordPress-Android/blob/feature/wp-stories/alpha2/WordPress/src/main/java/org/wordpress/android/util/KeyboardResizeViewUtil.java). 

The bug that's happening now is when we navigate to the tags option we are getting an [IllegalArgumentException](https://stackoverflow.com/questions/7100555/preventing-catching-illegalargumentexception-parameter-must-be-a-descendant-of/12888761#12888761).

<kbd><img src="https://user-images.githubusercontent.com/1509205/88045968-f9dcc100-cb14-11ea-930f-afeafc9a4727.gif" width="320"></kbd>

```
 java.lang.IllegalArgumentException: parameter must be a descendant of this view
        at android.view.ViewGroup.offsetRectBetweenParentAndChild(ViewGroup.java:6143)
        at android.view.ViewGroup.offsetDescendantRectToMyCoords(ViewGroup.java:6072)
        at android.view.ViewRootImpl.scrollToRectOrFocus(ViewRootImpl.java:3920)
        at android.view.ViewRootImpl.draw(ViewRootImpl.java:3493)
        at android.view.ViewRootImpl.performDraw(ViewRootImpl.java:3418)
        at android.view.ViewRootImpl.performTraversals(ViewRootImpl.java:2755)
        at android.view.ViewRootImpl.doTraversal(ViewRootImpl.java:1721)
        at android.view.ViewRootImpl$TraversalRunnable.run(ViewRootImpl.java:7598)
```

### At first I thought the below points were causing the issue : 


So far based on my research it seems we there's some inconsistency taking place with the `EditText` on the `TagsFragment` and how it receives focus because sometimes the bug doesn't present itself, while other times it causes a crash. 

I have tried. 

* Clearing the focus of the `EditText`
* Clearing the full-screen mode window flag programmatically. 

These didn't work so looking to hear your thoughts.

### Approach after validating the above doesn't matter by commenting out that code

So it seems, the issue could be that we have the keyboard open while changing fragments, so that could be causing the exception to occur. Still investigating.

An idea I am now playing with to move this functionality forward is if we would focus the `EditText` on the `TagsFragment` but close the keyboard before doing the navigation. Only caveat of this would be the would have to click the keyboard open, tapping the edit text. 

## Reviewing 
Only 1 reviewer is needed but anyone can review. 

## Submitter Checklist
